src/Raven.Server/Documents/Queries/QueryMetadata.cs

```diff
@@ -117,33 +118,92 @@ private void Build(BlittableJsonReaderObject parameters)
             {
-                OrderBy = new(string Name, OrderByFieldType OrderingType, bool Ascending)[Query.OrderBy.Count];
+                HandleOrderByClause(parameters);
+            }
 
-                for (var i = 0; i < Query.OrderBy.Count; i++)
-                {
-                    var order = Query.OrderBy[i];
-                    var indexFieldName = GetIndexFieldName(QueryExpression.Extract(QueryText, order.Expression.Field));
+            if (Query.SelectFunctionBody != null)
+                HandleSelectFunctionBody();
+            else if (Query.Select != null)
+                FillSelectFields(parameters);
 
-                    switch (order.Expression.Type)
-                    {
-                        case OperatorType.Method:
-                            OrderBy[i] = ExtractOrderByFromMethod(order, indexFieldName, parameters);
-                            break;
-                        case OperatorType.Field:
-                            OrderBy[i] = (indexFieldName, order.FieldType, order.Ascending);
-                            break;
-                        default:
-                            ThrowInvalidOperatorTypeInOrderBy(order.Expression.Type, QueryText, parameters);
-                            break;
-                    }
+        }
+
+        private void HandleSelectFunctionBody()
+        {
+            if (Query.Select != null && Query.Select.Count > 0)
+                ThrowInvalidFunctionSelectWithMoreFields();
+
+            if (RootAliasPaths.Count == 0)
+                ThrowMissingAliasOnSelectFunctionBody();
+
+            var name = "__selectOutput";
+            if (Query.DeclaredFunctions != null &&
+                Query.DeclaredFunctions.ContainsKey(name))
+                ThrowUseOfReserveFunctionBodyMethodName();
+
+            var sb = new StringBuilder();
+
+            sb.Append("function ").Append(name).Append("(");
+            int index = 0;
+            var args = new SelectField[RootAliasPaths.Count];
+
+            foreach (var alias in RootAliasPaths)
+            {
+                if (index != 0)
+                    sb.Append(", ");
+                sb.Append(alias.Key);
+                args[index++] = SelectField.Create(string.Empty, null, alias.Value,
+                    alias.Key.EndssWith("[]"), true);
+            }
+            sb.AppendLine(") { ");
+            sb.Append("    return ");
+
+            sb.Append(QueryExpression.Extract(Query.QueryText, Query.SelectFunctionBody));
+
+            sb.AppendLine(";").AppendLine("}");
+
+            if (Query.TryAddFunction(name, sb.ToString()) == false)
+                ThrowUseOfReserveFunctionBodyMethodName();
+
+
+            SelectFields = new[] { SelectField.CreateMethodCall(name, null, args) };
+        }
+
+        private static void ThrowUseOfReserveFunctionBodyMethodName()
+        {
+            throw new InvalidQueryException("When using select function body, the '__selectOutput' function is reserved");
```

Comment:
This overload of `InvalidQueryException` does not include the query text and parameters. Its existence might be confusing however we need it to properly deserialize the exception on the client side. 

We have `InvalidQueryException(string message, string queryText, BlittableJsonReaderObject parameters)` to throw on the server.